### PR TITLE
Azure provider support

### PR DESCRIPTION
**Pull request by sebbrandt87**
- addons api v1beta1
- azuredisk-csi-driver
- azurediskprovisioner
- all other usable addons activated

### CODE REVIEW

File: templates/azuredisk-csi-driver.yaml
Comment (shaneutt) [inline templates/azuredisk-csi-driver.yaml]: We have a temporarily required annotation for Addons in this repository to be compatible with Catalog:

```suggestion
  annotations:
    catalog.kubeaddons.mesosphere.io/addon-revision: "0.4.0-1"
```

File: templates/azurediskprovisioner.yaml
Comment (shaneutt) [inline templates/azurediskprovisioner.yaml]:
```suggestion
  annotations:
    catalog.kubeaddons.mesosphere.io/addon-revision: "1.0.0-1"
```

File: templates/localvolumeprovisioner.yaml
Comment (shaneutt) [inline templates/localvolumeprovisioner.yaml]: :+1: comments++

**Review by shaneutt — CHANGES_REQUESTED** [review]
Looks good, just a couple missing annotations.

File: templates/azuredisk-csi-driver.yaml
Comment (s12chung) [inline templates/azuredisk-csi-driver.yaml]: think you can remove this, as `aws` will never use this?

File: templates/azurediskprovisioner.yaml
Comment (s12chung) [inline templates/azurediskprovisioner.yaml]: think you can remove this too.

File: templates/azuredisk-csi-driver.yaml
Comment (sebbrandt87) [reply templates/azuredisk-csi-driver.yaml]: I want to be as explicit as possible ;)

File: templates/azurediskprovisioner.yaml
Comment (sebbrandt87) [reply templates/azurediskprovisioner.yaml]: I want to be as explicit as possible ;)

**Review by s12chung — CHANGES_REQUESTED** [review]
double checked, so there's 3 states for providers when calling `konvoy init`

1. not showing in `cluster.yaml`, when there is no `cloudProvider` entry
2. showing in `cluster.yaml`, but not enabled, when there is a `cloudProvider` entry, but enabled is not there or `false`
3. `enabled` in `cluster.yaml`,  when there is a `cloudProvider` entry and `enabled` is `true`

so it'd be better for those cloudProvider entries to be deleted.

https://github.com/mesosphere/kubeaddons/blob/74de967242d8d5e9a1e2103c335df4a3de0f8c2b/pkg/kubeaddons/v1beta1/utils.go#L134-L148

**Review by shaneutt — DISMISSED** [review]

**Review by shaneutt — APPROVED** [review]

**Review by s12chung — APPROVED** [review]

**Comment by hectorj2f** [comment]
@s12chung  We shouldn't have merged this PR before getting the approval to other related PRs for Azure :/. Anyway let's keep it IN for now :).

**Comment by s12chung** [comment]
Ah... my bad, saw it was ready...

**Pull request merged**
